core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java

```diff
@@ -648,11 +648,20 @@ public static String getShortBlockId(String blockId) {
             .replace(DATA_PART_PREFIX, "").replace(CARBON_DATA_EXT, "");
     // to remove compressor name
     if (!blockId.equalsIgnoreCase(blockIdWithCompressorName)) {
-      int index = blockIdWithCompressorName.lastIndexOf(".");
+      int index = blockIdWithCompressorName.lastIndexOf(POINT);
+      int fileSeperatorIndex = blockIdWithCompressorName.lastIndexOf(File.separator);
       if (index != -1) {
-        String replace =
-            blockIdWithCompressorName.replace(blockIdWithCompressorName.substring(index), "");
-        return replace;
+        String modifiedBlockId;
```

Comment:
Can extract common code from partition table and non-partition table to get block Id to new method

Reply:
We could use the same method for both partition and non-partition case. So removed `getShortBlockIdForPartitionTable `method.